aim/sdk/run.py

```diff
@@ -320,7 +321,22 @@ def track(
 
         Appends the tracked value to metric series specified by `name` and `context`.
         """
+
+        # since worker might be lagging behind, we want to log the timestamp of run.track() call,
+        # not the actual implementation execution time.
         track_time = time()
+        self.repo.tracking_queue.register_task(self._track_impl, value, track_time, name, step, epoch, context=context)
```

Comment:
This is a great point. Same applies to `context` I guess. I ust concerned about doing this in every case 🤔.

Reply:
Deep-copying is going to be okay in most of the cases when the data is not too big.
Otherwise, we can think of ways to make such heavy objects to be immutable. The class `Context` was supposed to be one initially, for the same reason.